cmd/govim: investigate mapping_defaults_go_to_def test flake

Investigate a mapping_defaults_go_to_def test flake seen in #414 (PR unrelated to flake).

```
        --- FAIL: TestScripts/scripts/mapping_defaults_go_to_def (0.57s)
            testscript.go:308: 
                # Test that the default mappings for GOVIMGoToDef work
                # Prelims; this does not affect out test of default mappings (0.007s)
                # Set initial position (0.129s)
                # Each block below is simply repeated for each of the default
                # mappings we have defined for GOVIMGoToDef and GOVIMGoToPrevDef
                # where we can test/trigger the mapping. For example, we can't
                # correctly trigger the mouse-based mappings because there isn't
                # the support within Vim to do so
                # Raw command calls (0.039s)
                # gd and <C-t> (0.041s)
                # <C-]> and <C-t> (0.022s)
                # Vim only, and at least v8.1.1262 (0.009s)
                # <C-LeftMouse> and <C-RightMouse> (0.111s)
                > vim ex 'call test_setmouse(3,15)'
                > vim ex 'call feedkeys(\"\\<C-LeftMouse>\", \"x\")'
                [stderr]
                unexpected command error: failed to ex(call feedkeys("\<C-LeftMouse>", "x")) in Vim: Caught 'got error whilst handling command:GOVIMGoToDef: failed to call gopls.Definition: no identifier found
                params were: &protocol.TextDocumentPositionParams{
                    TextDocument: protocol.TextDocumentIdentifier{URI:"file://$WORK/p.go"},
                    Position:     protocol.Position{Line:2, Character:12},
                }' in function <SNR>16_define[40]..<SNR>16_callbackCommand, line 5
                [exit status 255]
                FAIL: testdata/mapping_defaults_go_to_def.txt:50: unexpected command failure
```

Full build log

https://gist.github.com/myitcv/5b35695368ae105a6d7e6cdf891ebb46

--- a/cmd/govim/testdata/hover.txt
+++ b/cmd/govim/testdata/hover.txt
@@ -7,7 +7,7 @@
 [!vim:v8.1.1649] skip
 
 vim ex 'e main.go'
-vim ex 'call test_setmouse(6,13)'
+vim ex 'call test_setmouse(screenpos(bufwinid(\"main.go\"),6,13)[\"row\"],screenpos(bufwinid(\"main.go\"),6,13)[\"col\"])'
 vim ex 'call feedkeys(\"\\<MouseMove>\\<Ignore>\", \"xt\")'
 sleep 500ms
 vim -stringout expr 'GOVIM_internal_DumpPopups()'

--- a/cmd/govim/testdata/mapping_defaults_go_to_def.txt
+++ b/cmd/govim/testdata/mapping_defaults_go_to_def.txt
@@ -46,7 +46,7 @@ stdout '^\Q[3,15]\E$'
 [!vim:v8.1.1262] skip 'Need at least v8.1.1262 for test_setmouse'
 
 # <C-LeftMouse> and <C-RightMouse>
-vim ex 'call test_setmouse(3,15)'
+vim ex 'call test_setmouse(screenpos(bufwinid(\"p.go\"),3,15)[\"row\"],screenpos(bufwinid(\"p.go\"),3,15)[\"col\"])'
 vim ex 'call feedkeys(\"\\<C-LeftMouse>\", \"x\")'
 vim expr '[getcurpos()[1], getcurpos()[2]]'
 stdout '^\Q[4,7]\E$'
@@ -55,7 +55,7 @@ vim expr '[getcurpos()[1], getcurpos()[2]]'
 stdout '^\Q[3,15]\E$'
 
 # g<LeftMouse> and g<RightMouse>
-vim ex 'call test_setmouse(3,15)'
+vim ex 'call test_setmouse(screenpos(bufwinid(\"p.go\"),3,15)[\"row\"],screenpos(bufwinid(\"p.go\"),3,15)[\"col\"])'
 vim ex 'call feedkeys(\"g\\<LeftMouse>\", \"x\")'
 vim expr '[getcurpos()[1], getcurpos()[2]]'
 stdout '^\Q[4,7]\E$'

